test/e2e/operator_test.go

```diff
@@ -323,12 +332,21 @@ var _ = Describe("Operator API", func() {
 		})
 
 		AfterEach(func() {
+			By("Deleting the testing namespace")
 			Eventually(func() error {
-				err := client.Delete(clientCtx, ns)
-				if apierrors.IsNotFound(err) {
-					return nil
-				}
-				return err
+				return controllerclient.IgnoreNotFound(client.Delete(clientCtx, ns))
+			}).Should(Succeed())
+
+			By("Deleting the kiali-related CRDs")
+			Eventually(func() error {
+				return client.DeleteAllOf(clientCtx, &apiextensionsv1.CustomResourceDefinition{}, controllerclient.MatchingLabels{
+					fmt.Sprintf("operators.coreos.com/%s", operatorName.Name): "",
+				})
+			}).Should(Succeed())
+
+			By("Deleting the test Operator resource")
```

Comment:
It's highly likely that the generated Operator resource will be left behind, despite this check being in place, as there's some caching issues in the current operator controller. I added a top-level DeleteCollection call in the AfterEach block to help alleviate the frequency this happens.